code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -361,12 +360,7 @@
 
 /mob/living/silicon/robot/drone/update_canmove(delay_action_updates = 0)
 	. = ..()
-	if(emagged)
-		density = 1
-		if(world.time - emagged_time > EMAG_TIMER)
-			shut_down(TRUE)
-		return
-	density = 0 //this is reset every canmove update otherwise
+	density = emagged ? TRUE : FALSE //this is reset every canmove update otherwise
```

Comment:
Why not just do `density = emagged`? Both are booleans.

Reply:
changed